sycl/test-e2e/USM/properties.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/ext/intel/experimental/usm_properties.hpp>
+#include <sycl/usm/usm_allocator.hpp>
+
+int main() {
+  sycl::queue q;
+
+  // Ensure properties are supported when constructing the allocator:
+  sycl::usm_allocator<int, sycl::usm::alloc::shared> allocator{
+      q,
+      {sycl::ext::oneapi::property::usm::device_read_only{},
+       sycl::ext::intel::experimental::property::usm::buffer_location{1}}};
```

Comment:
There are some E2E tests for these properties (see for example Adapters/level_zero/usm_device_read_only.cpp and USM/buffer_location.cpp). Could these get some test cases with the allocator as well? Primarily I want to make sure the properties are being passed correctly to the allocations.